lib/ios/RNNNavigationStackManager.m

```diff
@@ -64,9 +64,10 @@ - (void)popToRoot:(UIViewController*)viewController animated:(BOOL)animated comp
 
 - (void)setStackRoot:(UIViewController *)newRoot fromViewController:(UIViewController *)fromViewController animated:(BOOL)animated completion:(RNNTransitionCompletionBlock)completion rejection:(RNNTransitionRejectionBlock)rejection {
 	UINavigationController* nvc = fromViewController.navigationController;
+	NSArray *newViewControllers = [[newRoot navigationController] viewControllers];
 	
 	[self performAnimationBlock:^{
-		[nvc setViewControllers:@[newRoot] animated:animated];
+		[nvc setViewControllers:newViewControllers ?: @[newRoot] animated:animated];
```

Comment:
Basically when `newViewControllers` has array it is when `setStackRoot` is called like this:
```
Navigation.setStackRoot(this.props.componentId, {
  stack: {
    children: [
      // children
    ],
  },
});
```

And when it is nil `setStackRoot` is called like this:
```
Navigation.setStackRoot(this.props.componentId, {
  component: {
    // component stuff
  },
});
```